client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -118,8 +120,47 @@ const announceFailure = ( { dispatch, getState }, { query: { siteId } } ) => {
 	dispatch( errorNotice( error ) );
 };
 
+export const fetchCommentsTree = ( { dispatch }, action ) => {
+	const { siteId, status = 'unapproved' } = action.query;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/sites/${ siteId }/comments-tree`,
+				apiNamespace: 'wpcom/v2',
+				query: {
+					status: ( 'unapproved' === status ) ? 'pending' : status,
+				},
+			},
+			action,
+		)
+	);
+};
+
+const addCommentsTree = ( { dispatch }, { query: { siteId, status } }, next, data ) => {
+	const [ commentsCount, commentsTree ] = data;
+
+	const count = parseInt( commentsCount[ 0 ] );
+
+	const tree = keyBy( map( commentsTree, comment => ( {
+		commentId: comment[ 0 ],
+		postId: comment[ 1 ],
+		commentParentId: comment[ 2 ],
```

Comment:
actually arrays are also objects and `get( [1, 2, 3], '[0]' )` will return `1`

even better? list destructuring…

```js
keyBy( map( commentsTree, ( [ commentId, postId, parentId ] ) => ( { commentId, postId, parentId } ) ) );
```